File: Programming Basics/C++/While loops/ExamPreparation/ExamPreparation.cpp
#include <iostream>
#include <string>
using namespace std;

int main()
{
	int maxBadGradesCount;
	cin >> maxBadGradesCount;

	int currentBadGradesCount = 0, sum = 0, count = 0;
	bool isSuccessful = true;

	string input, lastProblem;

	cin >> ws;
	getline(cin, input);
	while (input != "Enough") {
		int grade;
		cin >> grade;

		sum += grade;
		count++;
		lastProblem = input;

		if (grade <= 4) {
			currentBadGradesCount++;
			if (currentBadGradesCount == maxBadGradesCount) {
				isSuccessful = false;
				break;
			}
		}

		// NOTE: The previous if statement is equivalent to:
		// if (grade <= 4 && ++currentBadGradesCount == maxBadGradesCount) {
		// 	  isSuccessful = false;
		//	  break;
		// }

		cin >> ws;
		getline(cin, input);
	}

	if (isSuccessful) {
		cout.setf(ios::fixed);
		cout.precision(2);

		cout << "Average score: " << 1.0 * sum / count << endl;
		cout << "Number of problems: " << count << endl;
		cout << "Last problem: " << lastProblem << endl;
	}
	else {
		cout << "You need a break, " << currentBadGradesCount << " poor grades." << endl;
	}
}

File: Programming Basics/JavaScript/While loops/Exercises/exam-preparation.js
function solve(input) {
    const maxBadGradesCount = Number(input[0]);

    let currentBadGradesCount = 0;
    let sum = 0;
    let count = 0;
    let lastProblem = "";
    let isSuccessful = true;

    let index = 1;
    while (input[index] != "Enough") {
        const grade = Number(input[index + 1]);

        sum += grade;
        count++;
        lastProblem = input[index];

        if (grade <= 4) {
            currentBadGradesCount++;
            if (currentBadGradesCount === maxBadGradesCount) {
                isSuccessful = false;
                break;
            }
        }

        // NOTE: The previous if statement is equivalent to:
        // if (grade <= 4 && ++currentBadGradesCount === maxBadGradesCount) {
        // 	  isSuccessful = false;
        //	  break;
        // }

        index += 2;
    }

    if (isSuccessful) {
        console.log(`Average score: ${(sum / count).toFixed(2)}`);
        console.log(`Number of problems: ${count}`);
        console.log(`Last problem: ${lastProblem}`);
    } else {
        console.log(`You need a break, ${currentBadGradesCount} poor grades.`);
    }
}
